src/gwc/src/main/java/org/geoserver/gwc/GWC.java

```diff
@@ -762,8 +762,13 @@ public final ConveyorTile dispatch(
         }
 
         if (!tld.layerExists(layerName)) {
-            requestMistmatchTarget.append("not a tile layer");
-            return null;
+            // GEOS-9431-check in catalog also if not found TileLayerDispatcher
+            LayerInfo layerInfo = catalog.getLayerByName(layerName);
```

Comment:
This should indeed be using the LocalWorkspaceCatalog

Reply:
will LocalWorkspaceCatalog

Reply:
As in, the catalog reference you get from Spring should already contain LocalWorkspaceCatalog as one of the wrappers, don't think there is a need to be explicit.